modules/util/remove.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package util
+
+import (
+	"os"
+	"syscall"
+	"time"
+)
+
+// Remove removes the named file or (empty) directory with at most 5 attempts.Remove
+func Remove(name string) error {
+	var err error
+	for i := 0; i < 5; i++ {
+		err = os.Remove(name)
+		if err == nil {
+			break
+		}
+		unwrapped := err.(*os.PathError).Err
+		if unwrapped == syscall.EBUSY || unwrapped == syscall.ENOTEMPTY || unwrapped == syscall.EPERM {
+			// try again
+			<-time.After(100 * time.Millisecond)
+			continue
+		}
+
+		// TODO: Consider handling EMFILE?
```

Comment:
There is a slightly more up to date rimraf implementation at https://github.com/nodejs/node/blob/master/lib/internal/fs/rimraf.js which defineds `'EBUSY', 'EMFILE', 'ENFILE', 'ENOTEMPTY', 'EPERM'` as retryable. Maybe we should just copy that.

Reply:
EMFILE and ENFILE only occur on file open - can they really occur on deletion?

Reply:
If we only unlink single files, it should be safe. Just for recursive deletion that may involve readdir, they can happen.

Reply:
I think we should retry EMFILE and ENFILE for the `RemoveAll` case which does involve `stat`.

Reply:
Fine